clippy_lints/src/literal_representation.rs

```diff
@@ -529,19 +529,23 @@ impl LiteralRepresentation {
             then {
                 let digit_info = DigitInfo::new(&src, false);
                 if digit_info.radix == Radix::Decimal {
-                    let val = digit_info.digits
+                    if let Ok(val) = digit_info.digits
                         .chars()
                         .filter(|&c| c != '_')
                         .collect::<String>()
-                        .parse::<u128>().unwrap();
-                    if val < u128::from(self.threshold) {
-                        return
+                        .parse::<u128>() {
+                        if val < u128::from(self.threshold) {
+                            return
+                        }
+                        let hex = format!("{:#X}", val);
+                        let digit_info = DigitInfo::new(&hex[..], false);
```

Comment:
```suggestion
                        let digit_info = DigitInfo::new(&hex, false);
```
should also work, shouldn't it?